infrastructure/templates/template.json

```diff
@@ -1368,7 +1368,23 @@
     "searchReaderManagedIdentityName": "[concat(parameters('subscription'), '-', parameters('environment'), '-id-srch-reader')]",
     "searchServiceApiVersion": "2025-02-01-preview",
     "userAssignedIdentitiesApiVersion": "2024-11-30",
-    "storageAccountBackupContributorRoleId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'e5e2a7ff-d759-4cd2-bb51-3152d37e2eb1')]"
+    "storageAccountBackupContributorRoleId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'e5e2a7ff-d759-4cd2-bb51-3152d37e2eb1')]",
+    "baseAdminAllowedOrigins": [
+      "[concat('https://', parameters('adminUri'))]"
+    ],
+    "devAdminAllowedOrigins": [
+      "https://localhost:5021",
+      "http://localhost:5021"
+    ],
+    "conditionalDevAdminAllowedOrigins": "[if(equals(parameters('environmentName'), 'Development'), variables('devAdminAllowedOrigins'), createArray())]",
+    "basePublicAllowedOrigins": [
+      "[concat('https://', parameters('domain'))]",
+      "[concat('https://', variables('publicAppName'), '.azurewebsites.net')]"
+    ],
+    "devPublicAllowedOrigins": [
+      "http://localhost:3000"
+    ],
+    "conditionalDevPublicAllowedOrigins": "[if(equals(parameters('environmentName'), 'Development'), variables('devPublicAllowedOrigins'), createArray())]"
```

Comment:
Cool. I think I would have avoided making the template having to know anything about 'dev' specifically, by trying  something like a `adminAllowedOrigins` parameter which defaults to an empty array, and allowing different environment param files like `dev.parameters.json` define their own array of allowed origins.

Same for `publicAllowedOrigins`.

The template would then just union `adminAllowedOrigins` with `baseAdminAllowedOrigins`, and union `publicAllowedOrigins` with `basePublicAllowedOrigins`.

Then the environment param files do their job and it avoids cluttering the template with references to specific environments.

I think this is ok if you want to leave it like this. Plus without checking I'm not sure you can have parameters which are arrays!